src/util/cron.c

```diff
@@ -177,15 +214,45 @@ void Cron_Stop(void) {
 	cron = NULL;
 }
 
-void Cron_AddTask(uint when, CronTaskCB cb, void *pdata) {
+CronTaskHandle Cron_AddTask(uint when, CronTaskCB cb, void *pdata) {
 	ASSERT(cron != NULL);
-	ASSERT(cb != NULL);
+	ASSERT(cb   != NULL);
 
 	CRON_TASK *task = rm_malloc(sizeof(CRON_TASK));
 	task->cb     =  cb;
 	task->pdata  =  pdata;
 	task->due    =  due_in_ms(when);
+	task->state  =  TASK_PENDING;
 
 	CRON_InsertTask(task);
+
+	return (uintptr_t)task;
+}
+
+void Cron_AbortTask(CronTaskHandle t) {
+	ASSERT(cron != NULL);
+
+	CRON_TASK *task = (CRON_TASK *)t;
+
+	pthread_mutex_lock(&cron->mutex);
```

Comment:
lock all queries for long time

Reply:
I don't think so, (given the current tasks we're issuing)
In addition it would hold this lock for a "long" period of time, when the caller of this function is aborting when the task is executing, the chances of that are not high.